### PR TITLE
Fix CifData and tcod exporters for python3

### PR DESCRIPTION
**Pull request by ltalirz**
* select working version of PyCifRW in `setup_requirements.py` for python2 and python3

### CODE REVIEW

**Comment by ltalirz** [comment]
@merkys I've fixed the formatter issues now - In order to enable automatic code formatting + syntax checking, just do
```
pip install -e .[dev_precommit]
pre-commit install
```

**Comment by merkys** [comment]
Work finished. @giovannipizzi you may review the code now.

File: aiida/backends/tests/dbimporters.py [hunk 1]
```diff
@@ -202,17 +205,15 @@ def test_dbentry_creation(self):
         results = PcodSearchResults([{'id': '12345678'}])
         self.assertEquals(len(results), 1)
         self.assertEquals(str(results.at(0)),
-                          'PcodEntry(license="CC0",'
-                          'db_name="Predicted Crystallography Open Database",'
-                          'version=None,'
+                          'PcodEntry(db_name="Predicted Crystallography Open Database",'
```
Comment (giovannipizzi) [inline aiida/backends/tests/dbimporters.py]: Do we really want to compare the string representation? I think this is fragile. Better to compare specific values (e.g. `assertEquals(results.at(0).db_name, "Predicted..."` or something equivalent)

Comment (giovannipizzi) [reply aiida/backends/tests/dbimporters.py]: This is valid for many other places

File: aiida/tools/dbexporters/tcod.py [hunk 2]
```diff
@@ -200,21 +202,21 @@ def encode_textfield_quoted_printable(content):
...
         if 'prefix' in m.groupdict().keys():
             prefix = m.group('prefix')
         if 'postfix' in m.groupdict().keys():
             postfix = m.group('postfix')
         h = hex(ord(m.group('chr')))[2:].upper()
         if len(h) == 1:
-            h = "0{}".format(h)
-        return "{}={}{}".format(prefix, h, postfix)
+            h = "0%s" % h
+        return b"%s=%s%s" % (prefix, h.encode('utf-8'), postfix)
```
Comment (giovannipizzi) [inline aiida/tools/dbexporters/tcod.py]: Is there a reason to use % instead of .format()? In all the rest of the code we use .format()

Comment (giovannipizzi) [reply aiida/tools/dbexporters/tcod.py]: (Also for this, same question in many points)

File: aiida/backends/tests/dbimporters.py (hunk 1 above)
Comment (merkys) [reply aiida/backends/tests/dbimporters.py]: These string representations should be pretty stable. Unless new properties are added, they should stay completely identical.

File: aiida/tools/dbexporters/tcod.py (hunk 2 above)
Comment (merkys) [reply aiida/tools/dbexporters/tcod.py]: @yakutovicha, have you changed these lines? I am actually not aware of the differences between ``%`` and ``format``.

File: aiida/backends/tests/tcodexporter.py [hunk 3]
```diff
@@ -567,28 +563,28 @@ def check_gzip_base64(self, text):
...
-        check_quoted_printable(self, 'line\n;line', 'line\n=3Bline')
-        check_quoted_printable(self, 'tabbed\ttext', 'tabbed=09text')
-        check_quoted_printable(self, 'angstrom Å', 'angstrom =C3=85')
-        check_quoted_printable(self, 'line\rline\x00', 'line=0Dline=00')
+        check_ncr(self, b'.', b'&#46;')
+        check_ncr(self, b'?', b'&#63;')
+        check_ncr(self, b';\n', b'&#59;\n')
+        check_ncr(self, b'line\n;line', b'line\n&#59;line')
+        check_ncr(self, b'tabbed\ttext', b'tabbed&#9;text')
+        check_ncr(self, u'angstrom Å'.encode('utf-8'), b'angstrom &#195;&#133;')
```
Comment (merkys) [inline aiida/backends/tests/tcodexporter.py]: Add a comment here in order to explain why we have two bytes in return.

File: aiida/tools/dbexporters/tcod.py [hunk 4]
```diff
@@ -272,9 +274,12 @@ def decode_textfield_ncr(content):
     import re
 
     def match2str(m):
-        return chr(int(m.group(1)))
+        if isinstance(m.group(1), str):
```
Comment (merkys) [inline aiida/tools/dbexporters/tcod.py]: Use ``six.PY2`` instead. Document it well.

Comment (merkys) [reply aiida/tools/dbexporters/tcod.py]: Fixed and documented.

File: aiida/backends/tests/tcodexporter.py (hunk 3 above)
Comment (merkys) [reply aiida/backends/tests/tcodexporter.py]: Comment added.

File: aiida/backends/tests/dbimporters.py (hunk 1 above)
Comment (merkys) [reply aiida/backends/tests/dbimporters.py]: Converted to dictionary comparison. Should be more stable.

File: aiida/tools/dbexporters/tcod.py (hunk 2 above)
Comment (merkys) [reply aiida/tools/dbexporters/tcod.py]: > (Also for this, same question in many points)

This seems to be the last occurrence now, I have replaced others.

File: aiida/tools/dbexporters/tcod.py [hunk 5]
```diff
@@ -268,16 +267,27 @@ def decode_textfield_ncr(content):
...
+
+        :param match: match result of re.sub
+        :return: a single byte having a value of the first group in re.sub
+        """
+        byte_value = int(m.group(1))
+        if six.PY2:
+            return chr(byte_value)
         else:
-            return bytes([int(m.group(1))])
+            return bytes([byte_value])
```
Comment (giovannipizzi) [inline aiida/tools/dbexporters/tcod.py]: Just realized that six has the int2byte function:
https://pythonhosted.org/six/#six.int2byte
Better to use this

Comment (merkys) [reply aiida/tools/dbexporters/tcod.py]: Fixed, thanks!

**Pull request merged**